mavros_extras/src/plugins/odom.cpp

```diff
@@ -66,87 +67,97 @@ class OdometryPlugin : public plugin::PluginBase {
-
-		ftf::EigenMapConstCovariance6d cov_vel_map(cov_vel.data());
-		ROS_DEBUG_STREAM_NAMED("odom", "Odometry: velocity covariance matrix: " << std::endl << cov_vel_map);
+		// lookup transforms
+		Eigen::Affine3d tfChild2NED;
+		Eigen::Affine3d tfParent2NED;
+		try {
+			tfChild2NED = tf2::transformToEigen(m_uas->tf2_buffer.lookupTransform(
+				odom->child_frame_id, "local_origin_ned", ros::Time(0)));
+			tfParent2NED = tf2::transformToEigen(m_uas->tf2_buffer.lookupTransform(
+				odom->header.frame_id, "local_origin_ned", ros::Time(0)));
+		} catch (tf2::TransformException &ex) {
+			ROS_WARN("%s",ex.what());
+			ros::Duration(1.0).sleep();
+			return;
+		}
+
+		// unpack data into Eigen
+		Eigen::Vector3d posParent(
+			odom->pose.pose.position.x,
+			odom->pose.pose.position.y,
+			odom->pose.pose.position.z);
+
+		Eigen::Vector3d linVelChild(
+			odom->twist.twist.linear.x,
+			odom->twist.twist.linear.y,
+			odom->twist.twist.linear.z);
+
+		Eigen::Vector3d angVelChild(
+			odom->twist.twist.angular.x,
+			odom->twist.twist.angular.y,
+			odom->twist.twist.angular.z);
+
+		Eigen::Quaterniond qParent2Child(
+			odom->pose.pose.orientation.w,
+			odom->pose.pose.orientation.x,
+			odom->pose.pose.orientation.y,
+			odom->pose.pose.orientation.z);
+
+		Eigen::Vector3d linVelNED;
+		Eigen::Vector3d angVelNED;
+		Eigen::Vector3d posNED;
+		Eigen::Quaterniond qParent2NED;
+
+		Eigen::Matrix<double, 6, 6> poseCovParent;
+		Eigen::Matrix<double, 9, 9> poseCovNED;
+		Eigen::Matrix<double, 6, 6> twistCovChild;
+		Eigen::Matrix<double, 3, 3> attCovNED;
+
+		for (int i=0; i<6; i++) for (int j=0; j<6; j++) {
+			poseCovParent(i, j) = odom->pose.covariance[i*6 + j];
+			twistCovChild(i, j) = odom->pose.covariance[i*6 + j];
+		}
+
+		// apply transforms
+		linVelNED = tfChild2NED.linear()*linVelChild;
+		angVelNED = tfChild2NED.linear()*angVelChild;
+		posNED = tfParent2NED.linear()*posParent;
+		qParent2NED = tfParent2NED.linear()*qParent2Child;
+
+		// constrauct 9x6 matrix to rotate pose from parent to NED
+		// note it is 9x6 since the lpos matrix has accel states
+		// while ROS odometry message doesn't
+		Eigen::Matrix<double, 9, 6> RPose;
+		RPose.block<3, 3>(0, 0) = tfParent2NED.linear();
+		RPose.block<3, 3>(3, 3) = tfParent2NED.linear();
 
 		// get timestamp
 		uint64_t stamp_usec = odom->header.stamp.toNSec() / 1e3;
-		const auto zerov3f = Eigen::Vector3f::Zero();
 
 		/* -*- LOCAL_POSITION_NED_COV parser -*- */
 		mavlink::common::msg::LOCAL_POSITION_NED_COV lpos {};
 
 		lpos.time_usec = stamp_usec;
+		poseCovNED = RPose*poseCovParent*RPose.transpose();
 		lpos.estimator_type = utils::enum_value(estimator_type);
-		ftf::covariance9d_urt_to_mavlink(cov_full_tf, lpos.covariance);
+		for (int i=0; i<6; i++) for (int j=0; j<6; j++) {
+			lpos.covariance[i*6 + j] = poseCovNED(i, j);
+		}
 
 		// [[[cog:
-		// for a, b in (('', 'pos_ned'), ('v', 'lin_vel_ned'), ('a', 'zerov3f')):
+		// for a, b in (('', 'posNED'), ('v', 'linVelNED')):
```

Comment:
is there a reason you changed from `pos_ned` to `posNED`? Throughout MAVROS we use the `name_name` convention instead of `nameName`. Would be good to keep that consistency here if you don't mind.